### ISSUE TITLE
Linux version

### ISSUE THREAD

**Issue by probonopd** [OpenNewsLabs/autoEdit_2#36]
Is there any reason not to publish a Linux version?

**Comment by pietrop** [comment]
Hi @probonopd, 

I am looking into both a windows and a  Linux version. 

In order to do so there as need to : 

1. Add ffmpeg and ffprobe binaries specific to these systems.which I have done using the [static ffmpeg](https://www.npmjs.com/package/ffmpeg-static) and [static ffprobe](https://github.com/joshwnj/ffprobe-static) but needs testing. 

2. Also it needs going through the code base and changing any accourance of `/` in file paths across the project to the node `path` library to make it cross platform compatible.

3. It needs compiling into Linux and testing on a linux machine (or virtual machine) to see what else if anything would need adjusting, by looking at the errors in the console if any.

4. I don't have a windows or a Linux machine, so I am setting up virtual box, but it's not as fast of a workflow. 

I'd be curious to learn more about your use case tho, what would you be using autoEdit for?

**Comment by probonopd** [comment]
Hi @pietrop I don't have a use case yet, just wanted to try it out. Instead of setting up your own Linux build machine, you could also use a service like [Travis CI](https://travis-ci.org/). I use it a lot for my work on [AppImages](http://appimage.org/). Or, you could use the [Open Build Service](https://build.opensuse.org/).

**Comment by pietrop** [comment]
Ah, that sounds interesting, will need to look into that when I circle back to this. 

For linux which distribution would work best for you?

**Comment by sedubois** [comment]
+1 for a Windows version! I have macOS but collaborate with a colleague who uses Windows.

**Comment by pietrop** [comment]
@sedubois thanks for this input, quick question to you both, would a web app be just as good or are there preferences on it running as a desktop app?

**Comment by sedubois** [comment]
I think a web app would be more flexible, wouldn't it? No need for cumbersome installations etc. The question is more on your side, how do you keep track of heavy files etc.

**Comment by sedubois** [comment]
An API + web app would also make it easier for people to collaborate together.

**Comment by pietrop** [comment]
That's a good point, the reason why I started with a desktop app is that if you are working with files from a camcorder, in this setup you could add them to autoEdit straight from the card (using ffmpeg to make audio and video preview and ffprobe to read metadata info for the EDL)

A web app has the trade of that it could be mobile responsive (if you check the [demo](autoedit.io/demo) on the phone, the front end already is) and work across devices.  But it might need to have limits on file size that can be uploaded. 

Altho for example it just blew my mind seeing that [google drive can let you upload up to 5TB in size](https://support.google.com/drive/answer/37603?hl=en), I wonder how long that would take to upload, and what are the feasibility of that scenario. 

Ideally you'd have both desktop and web app, and they sync to the same account. So that you can choose what to use based on the circumstances. In this scenario, the original media, often huge in size does not sync, just the metadata to reconnect the EDL and the audio and video previews("proxies") if that makes sense.

Something to think about tho, is that the web app altho has costs for keeping it up and running, especially if it needs a backend, and is therefore harder to keep it freely available.

**Comment by sedubois** [comment]
@pietrop I have the feeling that going for several types of clients would bring more overhead, better to concentrate on one access point which is useful to as many people as possible. So the web app looks like the better way, IMHO. For sure as you mentioned it's possible to manage even very big files in this way, like the web client for Google Drive shows. I guess the files just need to be downloaded to the local machine before being manipulated, and re-uploaded as a new version once your editing is done. Maybe something like [GitHub LFS](https://git-lfs.github.com/) could allow to transfer only the file diffs.

Also as this is open-source software, I think that hosting costs is one of the things that you needn't worry about. With an OSS project it's normally possible to do everything for free. E.g the frontend or microservices could be dynamically deployed with [Zeit Now](https://zeit.co/now) or frontend could also be statically deployed for free with Netlify. I also like GraphQL and [Graphcool](https://www.graph.cool/) hosts open-source projects for free. And there's also plenty of free stuff for testing. Etc.

**Comment by sqr** [comment]
I must say a Linux or Windows version would be super useful to me, I work in a newsroom where people use both OS X and Windows. I personally don't own any OS X machines so unfortunately I can't try the software, it seems to be very promising!

**Comment by jsdephillips** [comment]
Hi @pietrop  I am also interested in a version I can run on PC. I'm working on a documentary and came across Auto Edit. It seems to be a perfect tool for the type of project I am working on in AVID Media Composer. I have hours of interviews that I am using to build the narrative of a documentary.  We have tried Auto Edit  on a Mac and it works wonderfully. Unfortunately, I have PC in my edit suite.  Web based has some drawbacks for me as some of the interview files I have are upward of 12 gigs. So upload time would be an issue. Please let me know when you have a PC version I can try out. Thank you so much. Jacquelyn

**Comment by sedubois** [comment]
Maybe it could be one JS codebase delivered on desktop with Electron.

**Comment by pietrop** [comment]
Hi @jsdephillips, 
Thanks for your message, glad you found it helpful, "digital paper-editing of docs" is the main use case I built autoEdit for. [On that note you might find this interesting, some notes on the underlying analogue workflow](https://pietropassarelli.gitbooks.io/how-to-tell-compelling-stories-out-of-video-inter/content/).

[As mentioned in previous comment](https://github.com/OpenNewsLabs/autoEdit_2/issues/36#issuecomment-311736613) these are the things that would be needed to make the Linux and Windows versions. I've done the first one, but could use some help with the rest of the list. 

- [X] **Add ffmpeg and ffprobe binaries specific to Linux, and Windows systems**. Which I have done using the static ffmpeg and static ffprobe but needs testing.
- [ ] Test that this works.
- [ ] **Change path references across the code base**, in other words changing any instance of `/` in file paths across the project to the node path library to make it cross platform compatible.
- [ ] **It needs compiling for Linux** and testing on a linux machine (or virtual machine) to see what else if anything would need adjusting, by looking at the errors in the console if any. And raising github issues accordingly.
- [ ] **It needs compiling for Windows version** and testing on a windows machine (or virtual machine?) to see what else if anything would need adjusting, by looking at the errors in the console if any. and raising GitHub issues accordingly.

I'd be happy to guide anyone who wants to have a go at this, as I might not have the time to dedicate to it at the moment. 

I also don't have a Windows or a Linux machine, so I'd welcome help in figure out an effective workflow to work around this.

@sedubois @probonopd let me know if you have any thoughts, or would like to get involved.

**Comment by probonopd** [comment]
Hey, it's a start... running on Xubuntu 16.04.

![](https://user-images.githubusercontent.com/2480569/38415567-6b56c2c2-3981-11e8-8d9c-f670afea945e.png)

Is getting this Bluemix thing free?

**Comment by pietrop** [comment]
Thanks @probonopd !

Great News about the linux version!

**re- IBM Bluemix**
Fair enough, let me see if I can find a linux user, I have asked a friend who as Linux machine if he can test it out, as well as [given a shout out on twitter](https://twitter.com/pietropassarell/status/982206350271823872). 
But perhaps @sqr might be interested in trying out?

**re-tests**
I am going to have a look as soon as I have a second, to try and get some clarity on what is causing that, but disabling the tests seems fine for now. 

**re-changing the path**
Same as above, I can do a quick pass through the code as soon as I get a sec, and then test the app out before commit to master.  Altho if anybody else wants to make a pull request. 
it's a matter of going through [`lib` folder](https://github.com/OpenNewsLabs/autoEdit_2/tree/master/lib).
- My feeling is that most of the backbone code won't have issues with that, but that it will be mostly int he [`interactive_transcription_generator`](https://github.com/OpenNewsLabs/autoEdit_2/tree/master/lib/interactive_transcription_generator) module. 

- This is [an example of how to do it using `path.join(`](https://github.com/OpenNewsLabs/autoEdit_2/blob/master/lib/interactive_transcription_generator/index.js#L116)
- _as I look through the code for an example of where to change it seems like it might be a solved issue but going to double check later_
-  essentially anywhere, where `/` to define a path is used, needs to be replaced with `path.join` so that on windows it can auto reverse the file path to `\`. 
- However I think this should not be an issue on on Linux version for now @probonopd.

Really excited seeing this coming together on other platforms!

**Comment by pietrop** [comment]
**re-linux version**
It seems like it works! Curiosity took over and I downloaded a [virtual box](https://www.virtualbox.org/) and [ubuntu](https://www.ubuntu.com)

Keyboard shortcuts also work, but need to use `ctrl` instead of `cmd`. 
- [ ] Need to update keyboard shortcuts menu to display `crtl` on linux instead of `cmd` accordingly. 
- [ ] remove pocketsphinx and gentle STT option while in linux os in new transcription view, as those option would currently work only on OSX. 

<img width="550" alt="ubuntu" src="https://user-images.githubusercontent.com/4661975/38449416-b2774a7a-3a06-11e8-8b5c-3363e5c5a9ec.png">

@probonopd I'd love to document how to set it up with Travis CLI and connect that to the autoEdit repo to automate the process. What's the best way to go about that?

For now I tried building the Linux version on  OSX, after looking at your fork and changing the `package.json` to be

```json
    "linux": {
      "category": "",
      "packageCategory": "GNOME;GTK;VideoEditing;PaperEditing;Transcriptions",
      "description": "autoEdit2 Desktop Client for Linux",
      "target": [
        "deb",
        "zip",
        "AppImage",
        "rpm"
      ],
```

to be 

```json
 "linux": {
      "category": "",
      "packageCategory": "GNOME;GTK;VideoEditing;PaperEditing;Transcriptions",
      "description": "autoEdit2 Desktop Client for Linux",
      "target": [
        "AppImage"
      ],
```

I then run 

```
 npm run build:linux
```

See Linux release here 

https://github.com/OpenNewsLabs/autoEdit_2/releases/tag/1.0.8

--- 
**re-test** It's [the tests in metadata reader that are causing the problem, I had hardcoded the path to a local video file.](https://github.com/probonopd/autoEdit_2/blob/patch-1/spec/video_metadata_reader/video_metadata_reader_spec.js#L11)
But I've seen that most of the rest of the test coverage has that problem, so need to figure out a better strategy for testing modules that have media files as inputs, perhaps using fixtures etc..
So I'd live it commented out for now..

---
**re-path** found an [example of something that needs changing as explained above for this to work on windows](https://github.com/OpenNewsLabs/autoEdit_2/blob/master/electron/watson_keys.js#L9)

**Comment by probonopd** [comment]
I can send a pull request for the Travis CI build automation if you like. It will build an A AppImage on each git push.

**Comment by pietrop** [comment]
That'll be great! thanks @probonopd !

I have a few questions tho:

- Is there a way for Travis CI to also do the same for the OSX version and the Windows version?
- Is there an easy way to change the script so that it will build an AppImage on each git push only if it's a new `version`? For example using the `package.json` version such as ` "version": "1.0.8",`

Thanks!

**Comment by pietrop** [comment]
another question re-`AppImage`

I found [here]( https://askubuntu.com/questions/774490/what-is-an-appimage-how-do-i-install-it
) that 

```
AppImages can be downloaded and run without installation or the need for root rights.

Making it executable
You can make the appImage executable as follows:

chmod a+x exampleName.AppImage
Executing it
You can execute an appImage as follows:

./exampleName.AppImage
```

My question is, if I run the command to make it executable before publishing the app `chmod a+x exampleName.AppImage` then the user doesn't need to do that on their end? and if that is the case is it possible to incorporated in the deployment script? So that the user only has to download and then double click on the `AppImage` to get up and running.

**Comment by pietrop** [comment]
Opened a new issue to talk about the Windows version specifics  https://github.com/OpenNewsLabs/autoEdit_2/issues/44

**Comment by probonopd** [comment]
The user needs to do this. This is how Linux security works. See https://discourse.appimage.org/t/how-to-make-an-appimage-executable/80

> Is there a way for Travis CI to also do the same for the OSX version and the Windows 

Yes for macOS, no for Windows (there is CircleCI for that)

> Is there an easy way to change the script so that it will build an AppImage on each git push only if it's a new version?

No.

**Comment by pietrop** [comment]
Ok great, thanks @probonopd !

For now how would I add macOS build to the Travis CI?

Is it as simple as adding the line below to the [`.travis.ym` config on line 35](https://github.com/probonopd/autoEdit_2/blob/f4eb9a634ae2870743dc1a54f4bf6a009f10923b/.travis.yml#L34) ?

```
- bash upload.sh ./dist/autoEdit2-*.dmg autoEdit2
```
---
Re-Pull request for the Travis CI build automation, sounds good, feel free to do that.

**Comment by pietrop** [comment]
@probonopd I tried and add your `.travis.yml` but got the following issue

https://travis-ci.org/OpenNewsLabs/autoEdit_2/builds/364009530

Did you had this problem as well when you tested it in your fork?

```
Worker information

hostname: 90be42b5-09ee-4c60-9646-060ec0cf8917@1.production-1-worker-org-a-6-gce

version: v3.6.0 https://github.com/travis-ci/worker/tree/170b2a0bb43234479fd1911ba9e4dbcc36dadfad

instance: travis-job-8387a485-db01-455b-9a8b-8ac48c11d289 travis-ci-garnet-trusty-1512502259-986baf0 (via amqp)

startup: 28.328544094s


mode of ‘/usr/local/clang-5.0.0/bin’ changed from 0777 (rwxrwxrwx) to 0775 (rwxrwxr-x)

system_info
Build system information

Build language: node_js

Build group: stable

Build dist: trusty

Build id: 364009530

Job id: 364009531

Runtime kernel version: 4.4.0-101-generic

travis-build version: ef2f5cbc9

Build image provisioning date and time

Tue Dec  5 19:58:13 UTC 2017

Operating System Details

Distributor ID:	Ubuntu

Description:	Ubuntu 14.04.5 LTS

Release:	14.04

Codename:	trusty

Cookbooks Version

7c2c6a6 https://github.com/travis-ci/travis-cookbooks/tree/7c2c6a6

git version

git version 2.15.1

bash version

GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)

gcc version

gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4

Copyright (C) 2013 Free Software Foundation, Inc.

This is free software; see the source for copying conditions.  There is NO

warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

docker version

Client:

 Version:      17.09.0-ce

 API version:  1.32

 Go version:   go1.8.3

 Git commit:   afdb6d4

 Built:        Tue Sep 26 22:42:38 2017

 OS/Arch:      linux/amd64

Server:

 Version:      17.09.0-ce

 API version:  1.32 (minimum version 1.12)

 Go version:   go1.8.3

 Git commit:   afdb6d4

 Built:        Tue Sep 26 22:41:20 2017

 OS/Arch:      linux/amd64

 Experimental: false

clang version

clang version 5.0.0 (tags/RELEASE_500/final)

Target: x86_64-unknown-linux-gnu

Thread model: posix

InstalledDir: /usr/local/clang-5.0.0/bin

jq version

jq-1.5

bats version

Bats 0.4.0

shellcheck version

0.4.6

shfmt version

v2.0.0

ccache version

ccache version 3.1.9

Copyright (C) 2002-2007 Andrew Tridgell

Copyright (C) 2009-2011 Joel Rosdahl

This program is free software; you can redistribute it and/or modify it under

the terms of the GNU General Public License as published by the Free Software

Foundation; either version 3 of the License, or (at your option) any later

version.

cmake version

cmake version 3.9.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).

heroku version

heroku-cli/6.14.39-addc925 (linux-x64) node-v9.2.0

imagemagick version

Version: ImageMagick 6.7.7-10 2017-07-31 Q16 http://www.imagemagick.org

md5deep version

4.2

mercurial version

Mercurial Distributed SCM (version 4.2.2)

(see https://mercurial-scm.org for more information)

Copyright (C) 2005-2017 Matt Mackall and others

This is free software; see the source for copying conditions. There is NO

warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

mysql version

mysql  Ver 14.14 Distrib 5.6.33, for debian-linux-gnu (x86_64) using  EditLine wrapper

openssl version

OpenSSL 1.0.1f 6 Jan 2014

packer version

Packer v1.0.2

Your version of Packer is out of date! The latest version

is 1.1.2. You can update by downloading from www.packer.io

postgresql client version

psql (PostgreSQL) 9.6.6

ragel version

Ragel State Machine Compiler version 6.8 Feb 2013

Copyright (c) 2001-2009 by Adrian Thurston

subversion version

svn, version 1.8.8 (r1568071)

   compiled Aug 10 2017, 17:20:39 on x86_64-pc-linux-gnu

Copyright (C) 2013 The Apache Software Foundation.

This software consists of contributions made by many people;

see the NOTICE file for more information.

Subversion is open source software, see http://subversion.apache.org/

The following repository access (RA) modules are available:

* ra_svn : Module for accessing a repository using the svn network protocol.

  - with Cyrus SASL authentication

  - handles 'svn' scheme

* ra_local : Module for accessing a repository on local disk.

  - handles 'file' scheme

* ra_serf : Module for accessing a repository via WebDAV protocol using serf.

  - using serf 1.3.3

  - handles 'http' scheme

  - handles 'https' scheme

sudo version

Sudo version 1.8.9p5

Configure options: --prefix=/usr -v --with-all-insults --with-pam --with-fqdn --with-logging=syslog --with-logfac=authpriv --with-env-editor --with-editor=/usr/bin/editor --with-timeout=15 --with-password-timeout=0 --with-passprompt=[sudo] password for %p:  --without-lecture --with-tty-tickets --disable-root-mailer --enable-admin-flag --with-sendmail=/usr/sbin/sendmail --with-timedir=/var/lib/sudo --mandir=/usr/share/man --libexecdir=/usr/lib/sudo --with-sssd --with-sssd-lib=/usr/lib/x86_64-linux-gnu --with-selinux

Sudoers policy plugin version 1.8.9p5

Sudoers file grammar version 43

Sudoers path: /etc/sudoers

Authentication methods: 'pam'

Syslog priority to use when user authenticates successfully: notice

Syslog priority to use when user authenticates unsuccessfully: alert

Send mail if the user is not in sudoers

Use a separate timestamp for each user/tty combo

Lecture user the first time they run sudo

Root may run sudo

Allow some information gathering to give useful error messages

Require fully-qualified hostnames in the sudoers file

Visudo will honor the EDITOR environment variable

Set the LOGNAME and USER environment variables

Length at which to wrap log file lines (0 for no wrap): 80

Authentication timestamp timeout: 15.0 minutes

Password prompt timeout: 0.0 minutes

Number of tries to enter a password: 3

Umask to use or 0777 to use user's: 022

Path to mail program: /usr/sbin/sendmail

Flags for mail program: -t

Address to send mail to: root

Subject line for mail messages: *** SECURITY information for %h ***

Incorrect password message: Sorry, try again.

Path to authentication timestamp dir: /var/lib/sudo

Default password prompt: [sudo] password for %p: 

Default user to run commands as: root

Value to override user's $PATH with: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin

Path to the editor for use by visudo: /usr/bin/editor

When to require a password for 'list' pseudocommand: any

When to require a password for 'verify' pseudocommand: all

File descriptors >= 3 will be closed before executing a command

Environment variables to check for sanity:

	TZ

	TERM

	LINGUAS

	LC_*

	LANGUAGE

	LANG

	COLORTERM

Environment variables to remove:

	RUBYOPT

	RUBYLIB

	PYTHONUSERBASE

	PYTHONINSPECT

	PYTHONPATH

	PYTHONHOME

	TMPPREFIX

	ZDOTDIR

	READNULLCMD

	NULLCMD

	FPATH

	PERL5DB

	PERL5OPT

	PERL5LIB

	PERLLIB

	PERLIO_DEBUG 

	JAVA_TOOL_OPTIONS

	SHELLOPTS

	GLOBIGNORE

	PS4

	BASH_ENV

	ENV

	TERMCAP

	TERMPATH

	TERMINFO_DIRS

	TERMINFO

	_RLD*

	LD_*

	PATH_LOCALE

	NLSPATH

	HOSTALIASES

	RES_OPTIONS

	LOCALDOMAIN

	CDPATH

	IFS

Environment variables to preserve:

	JAVA_HOME

	TRAVIS

	CI

	DEBIAN_FRONTEND

	XAUTHORIZATION

	XAUTHORITY

	PS2

	PS1

	PATH

	LS_COLORS

	KRB5CCNAME

	HOSTNAME

	HOME

	DISPLAY

	COLORS

Locale to use while parsing sudoers: C

Directory in which to store input/output logs: /var/log/sudo-io

File in which to store the input/output log: %{seq}

Add an entry to the utmp/utmpx file when allocating a pty

PAM service name to use

PAM service name to use for login shells

Create a new PAM session for the command to run in

Maximum I/O log sequence number: 0

Local IP address and netmask pairs:

	10.240.0.28/255.255.255.255

	172.17.0.1/255.255.0.0

Sudoers I/O plugin version 1.8.9p5

gzip version

gzip 1.6

Copyright (C) 2007, 2010, 2011 Free Software Foundation, Inc.

Copyright (C) 1993 Jean-loup Gailly.

This is free software.  You may redistribute copies of it under the terms of

the GNU General Public License <http://www.gnu.org/licenses/gpl.html>.

There is NO WARRANTY, to the extent permitted by law.

Written by Jean-loup Gailly.

zip version

Copyright (c) 1990-2008 Info-ZIP - Type 'zip "-L"' for software license.

This is Zip 3.0 (July 5th 2008), by Info-ZIP.

Currently maintained by E. Gordon.  Please send bug reports to

the authors using the web page at www.info-zip.org; see README for details.

Latest sources and executables are at ftp://ftp.info-zip.org/pub/infozip,

as of above date; see http://www.info-zip.org/ for other sites.

Compiled with gcc 4.8.2 for Unix (Linux ELF) on Oct 21 2013.

Zip special compilation options:

	USE_EF_UT_TIME       (store Universal Time)

	BZIP2_SUPPORT        (bzip2 library version 1.0.6, 6-Sept-2010)

	    bzip2 code and library copyright (c) Julian R Seward

	    (See the bzip2 license for terms of use)

	SYMLINK_SUPPORT      (symbolic links supported)

	LARGE_FILE_SUPPORT   (can read and write large files on file system)

	ZIP64_SUPPORT        (use Zip64 to store large files in archives)

	UNICODE_SUPPORT      (store and read UTF-8 Unicode paths)

	STORE_UNIX_UIDs_GIDs (store UID/GID sizes/values using new extra field)

	UIDGID_NOT_16BIT     (old Unix 16-bit UID/GID extra field not used)

	[encryption, version 2.91 of 05 Jan 2007] (modified for Zip 3)

Encryption notice:

	The encryption code of this program is not copyrighted and is

	put in the public domain.  It was originally written in Europe

	and, to the best of our knowledge, can be freely distributed

	in both source and object forms from any country, including

	the USA under License Exception TSU of the U.S. Export

	Administration Regulations (section 740.13(e)) of 6 June 2002.

Zip environment options:

             ZIP:  [none]

          ZIPOPT:  [none]

vim version

VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:43:18)

Included patches: 1-52

Extra patches: 8.0.0056

Modified by pkg-vim-maintainers@lists.alioth.debian.org

Compiled by buildd@

Huge version without GUI.  Features included (+) or not (-):

+acl             +farsi           +mouse_netterm   +syntax

+arabic          +file_in_path    +mouse_sgr       +tag_binary

+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static

-balloon_eval    +float           +mouse_urxvt     -tag_any_white

-browse          +folding         +mouse_xterm     -tcl

++builtin_terms  -footer          +multi_byte      +terminfo

+byte_offset     +fork()          +multi_lang      +termresponse

+cindent         +gettext         -mzscheme        +textobjects

-clientserver    -hangul_input    +netbeans_intg   +title

-clipboard       +iconv           +path_extra      -toolbar

+cmdline_compl   +insert_expand   -perl            +user_commands

+cmdline_hist    +jumplist        +persistent_undo +vertsplit

+cmdline_info    +keymap          +postscript      +virtualedit

+comments        +langmap         +printer         +visual

+conceal         +libcall         +profile         +visualextra

+cryptv          +linebreak       +python          +viminfo

+cscope          +lispindent      -python3         +vreplace

+cursorbind      +listcmds        +quickfix        +wildignore

+cursorshape     +localmap        +reltime         +wildmenu

+dialog_con      -lua             +rightleft       +windows

+diff            +menu            -ruby            +writebackup

+digraphs        +mksession       +scrollbind      -X11

-dnd             +modify_fname    +signs           -xfontset

-ebcdic          +mouse           +smartindent     -xim

+emacs_tags      -mouseshape      -sniff           -xsmp

+eval            +mouse_dec       +startuptime     -xterm_clipboard

+ex_extra        +mouse_gpm       +statusline      -xterm_save

+extra_search    -mouse_jsbterm   -sun_workshop    -xpm

   system vimrc file: "$VIM/vimrc"

     user vimrc file: "$HOME/.vimrc"

 2nd user vimrc file: "~/.vim/vimrc"

      user exrc file: "$HOME/.exrc"

  fall-back for $VIM: "/usr/share/vim"

Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1      

Linking: gcc   -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lgpm -ldl    -L/usr/lib/python2.7/config-x86_64-linux-gnu -lpython2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions      

iptables version

iptables v1.4.21

curl version

curl 7.35.0 (x86_64-pc-linux-gnu) libcurl/7.35.0 OpenSSL/1.0.1f zlib/1.2.8 libidn/1.28 librtmp/2.3

wget version

GNU Wget 1.15 built on linux-gnu.

rsync version

rsync  version 3.1.0  protocol version 31

gimme version

v1.2.0

nvm version

0.33.6

perlbrew version

/home/travis/perl5/perlbrew/bin/perlbrew  - App::perlbrew/0.80

phpenv version

rbenv 1.1.1-25-g6aa70b6

rvm version

rvm 1.29.3 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]

default ruby version

ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]

CouchDB version

couchdb 1.6.1

ElasticSearch version

5.5.0

Installed Firefox version

firefox 56.0.2

MongoDB version

MongoDB 3.4.10

PhantomJS version

2.1.1

Pre-installed PostgreSQL versions

9.2.24

9.3.20

9.4.15

9.5.10

9.6.6

RabbitMQ Version

3.6.14

Redis version

redis-server 4.0.6

riak version

2.2.3

Pre-installed Go versions

1.7.4

ant version

Apache Ant(TM) version 1.9.3 compiled on April 8 2014

mvn version

Apache Maven 3.5.2 (138edd61fd100ec658bfa2d307c43b76940a5d7d; 2017-10-18T07:58:13Z)

Maven home: /usr/local/maven-3.5.2

Java version: 1.8.0_151, vendor: Oracle Corporation

Java home: /usr/lib/jvm/java-8-oracle/jre

Default locale: en_US, platform encoding: UTF-8

OS name: "linux", version: "4.4.0-98-generic", arch: "amd64", family: "unix"

gradle version

------------------------------------------------------------

Gradle 4.0.1

------------------------------------------------------------

Build time:   2017-07-07 14:02:41 UTC

Revision:     38e5dc0f772daecca1d2681885d3d85414eb6826

Groovy:       2.4.11

Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015

JVM:          1.8.0_151 (Oracle Corporation 25.151-b12)

OS:           Linux 4.4.0-98-generic amd64

lein version

Leiningen 2.8.1 on Java 1.8.0_151 Java HotSpot(TM) 64-Bit Server VM

Pre-installed Node.js versions

v4.8.6

v6.12.0

v6.12.1

v8.9

v8.9.1

phpenv versions

  system

  5.6

* 5.6.32 (set by /home/travis/.phpenv/version)

  7.0

  7.0.25

  7.1

  7.1.11

  hhvm

  hhvm-stable

composer --version

Composer version 1.5.2 2017-09-11 16:59:25

Pre-installed Ruby versions

ruby-2.2.7

ruby-2.3.4

ruby-2.4.1


removed ‘/etc/apt/sources.list.d/basho_riak.list’

Network availability confirmed.

127.0.0.1 localhost nettuno travis vagrant

127.0.1.1 travis-job-8387a485-db01-455b-9a8b-8ac48c11d289 travis-job-8387a485-db01-455b-9a8b-8ac48c11d289 ip4-loopback trusty64

W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)

git.checkout
8.98s$ git clone --depth=50 --branch=master https://github.com/OpenNewsLabs/autoEdit_2.git OpenNewsLabs/autoEdit_2

Cloning into 'OpenNewsLabs/autoEdit_2'...

remote: Counting objects: 1192, done.

remote: Compressing objects: 100% (696/696), done.

remote: Total 1192 (delta 596), reused 946 (delta 457), pack-reused 0

Receiving objects: 100% (1192/1192), 84.16 MiB | 21.03 MiB/s, done.

Resolving deltas: 100% (596/596), done.

$ cd OpenNewsLabs/autoEdit_2

$ git checkout -qf b7813b432b126ac7a623dd5810235ab198d286a3


apt
Installing APT Packages (BETA)

$ export DEBIAN_FRONTEND=noninteractive

2.99s$ sudo -E apt-get -yq update &>> ~/apt-get-update.log

5.00s$ sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install libgnome-keyring-dev icnsutils

Reading package lists...

Building dependency tree...

Reading state information...

The following additional packages will be installed:

  gir1.2-gnomekeyring-1.0 libgnome-keyring-common libgnome-keyring0 libicns1

Suggested packages:

  gnome-keyring

The following NEW packages will be installed:

  gir1.2-gnomekeyring-1.0 icnsutils libgnome-keyring-common

  libgnome-keyring-dev libgnome-keyring0 libicns1

0 upgraded, 6 newly installed, 0 to remove and 175 not upgraded.

Need to get 160 kB of archives.

After this operation, 1,481 kB of additional disk space will be used.

Get:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 libgnome-keyring-common all 3.8.0-2 [5,536 B]

Get:2 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 libgnome-keyring0 amd64 3.8.0-2 [55.4 kB]

Get:3 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe amd64 libicns1 amd64 0.8.1-1 [30.9 kB]

Get:4 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 gir1.2-gnomekeyring-1.0 amd64 3.8.0-2 [5,898 B]

Get:5 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe amd64 icnsutils amd64 0.8.1-1 [17.4 kB]

Get:6 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 libgnome-keyring-dev amd64 3.8.0-2 [45.3 kB]

Fetched 160 kB in 2s (62.3 kB/s)

Selecting previously unselected package libgnome-keyring-common.

(Reading database ... 93696 files and directories currently installed.)

Preparing to unpack .../libgnome-keyring-common_3.8.0-2_all.deb ...

Unpacking libgnome-keyring-common (3.8.0-2) ...

Selecting previously unselected package libgnome-keyring0:amd64.

Preparing to unpack .../libgnome-keyring0_3.8.0-2_amd64.deb ...

Unpacking libgnome-keyring0:amd64 (3.8.0-2) ...

Selecting previously unselected package libicns1:amd64.

Preparing to unpack .../libicns1_0.8.1-1_amd64.deb ...

Unpacking libicns1:amd64 (0.8.1-1) ...

Selecting previously unselected package gir1.2-gnomekeyring-1.0.

Preparing to unpack .../gir1.2-gnomekeyring-1.0_3.8.0-2_amd64.deb ...

Unpacking gir1.2-gnomekeyring-1.0 (3.8.0-2) ...

Selecting previously unselected package icnsutils.

Preparing to unpack .../icnsutils_0.8.1-1_amd64.deb ...

Unpacking icnsutils (0.8.1-1) ...

Selecting previously unselected package libgnome-keyring-dev.

Preparing to unpack .../libgnome-keyring-dev_3.8.0-2_amd64.deb ...

Unpacking libgnome-keyring-dev (3.8.0-2) ...

Processing triggers for man-db (2.6.7.1-1ubuntu1) ...

Setting up libgnome-keyring-common (3.8.0-2) ...

Setting up libgnome-keyring0:amd64 (3.8.0-2) ...

Setting up libicns1:amd64 (0.8.1-1) ...

Setting up gir1.2-gnomekeyring-1.0 (3.8.0-2) ...

Setting up icnsutils (0.8.1-1) ...

Setting up libgnome-keyring-dev (3.8.0-2) ...

Processing triggers for libc-bin (2.19-0ubuntu6.13) ...

W: --force-yes is deprecated, use one of the options starting with --allow instead.


$ export PATH=./node_modules/.bin:$PATH

Updating nvm

nvm.install
3.71s$ nvm install 7

Downloading and installing node v7.10.1...

Downloading https://nodejs.org/dist/v7.10.1/node-v7.10.1-linux-x64.tar.xz...

Computing checksum with sha256sum

Checksums matched!

Now using node v7.10.1 (npm v4.2.0)


cache.1
Setting up build cache

$ export CASHER_DIR=$HOME/.casher

0.18s$ Installing caching utilities

0.00s

4.30sattempting to download cache archive

fetching master/cache-linux-trusty-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--node-7.tgz

fetching master/cache--node-7.tgz

could not download cache

0.00s

1.00sadding /home/travis/build/OpenNewsLabs/autoEdit_2/node_modules to cache

creating directory /home/travis/build/OpenNewsLabs/autoEdit_2/node_modules

adding /home/travis/.electron to cache

creating directory /home/travis/.electron

adding /home/travis/.npm to cache

adding /home/travis/.nvm to cache


$ node --version

v7.10.1

$ npm --version

4.2.0

$ nvm --version

0.33.8

install.1
0.01s$ node --version

v7.10.1


install.2
0.24s$ npm --version

4.2.0


1.50s$ npm install

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c: Cloning into bare repository '/home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c'...

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c: Permission denied (publickey).

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c: fatal: Could not read from remote repository.

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c: 

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c: Please make sure you have the correct access rights

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c: and the repository exists.

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: Cloning into bare repository '/home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638'...

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: Warning: Permanently added the RSA host key for IP address '192.30.253.112' to the list of known hosts.

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: Permission denied (publickey).

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: fatal: Could not read from remote repository.

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: 

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: Please make sure you have the correct access rights

npm ERR! git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffmpeg-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffmpeg-static-git-9dd1a638: and the repository exists.

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: appdmg@^0.4.5 (node_modules/appdmg):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for appdmg@0.4.5: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! Linux 4.4.0-101-generic

npm ERR! argv "/home/travis/.nvm/versions/node/v7.10.1/bin/node" "/home/travis/.nvm/versions/node/v7.10.1/bin/npm" "install"

npm ERR! node v7.10.1

npm ERR! npm  v4.2.0

npm ERR! code 128

npm ERR! Command failed: git clone --template=/home/travis/.npm/_git-remotes/_templates --mirror ssh://git@github.com/pietrop/ffprobe-static.git /home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c

npm ERR! Cloning into bare repository '/home/travis/.npm/_git-remotes/git-ssh-git-github-com-pietrop-ffprobe-static-git-7e62879c'...

npm ERR! Permission denied (publickey).

npm ERR! fatal: Could not read from remote repository.

npm ERR! 

npm ERR! Please make sure you have the correct access rights

npm ERR! and the repository exists.

40mnpm ERR! 

npm ERR! 

npm ERR! If you need help, you may report this error at:

npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:

npm ERR!     /home/travis/.npm/_logs/2018-04-09T08_28_36_605Z-debug.log

The command "npm install" failed and exited with 1 during .

Your build has been stopped.
```

**Comment by probonopd** [comment]
Sent PR at https://github.com/OpenNewsLabs/autoEdit_2/pull/45.

> Did you had this problem as well when you tested it in your fork?

Yes. See my PR that fixes it.

**Comment by pietrop** [comment]
Closing this as just integrated PR https://github.com/OpenNewsLabs/autoEdit_2/pull/45

For Windows related question, see separate issue https://github.com/OpenNewsLabs/autoEdit_2/issues/44
